### PR TITLE
Rubnews API

### PR DESCRIPTION
**Pull request by domai-tb**
This PR adds the backend for requesting news of the university.

It includes:
- Requesting the actual news feed from `news.rub.de/newsfeed` and parse them into news entities
- Local caching of these entities with the package [hive](https://pub.dev/packages/hive)
- Unit testing of all important backend classes
- Connection of Feed UI with Rubnews API to show news inside the feed page

### CODE REVIEW

File: lib/pages/rubnews/widgets/feed_item.dart [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import 'package:cached_network_image/cached_network_image.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 import 'package:animations/animations.dart';
```
Comment (C4s4r) [inline lib/pages/rubnews/widgets/feed_item.dart:4]: Again here: In order to keep the order _dart, flutter, external packages, internal imports_ this should be moved down.
```suggestion
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:animations/animations.dart';
import 'package:cached_network_image/cached_network_image.dart';
```

File: lib/utils/pages/feed_utils.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:campus_app/pages/rubnews/news_entity.dart';
+import 'package:campus_app/pages/rubnews/widgets/feed_item.dart';
+import 'package:campus_app/utils/pages/presentation_functions.dart';
+import 'package:flutter/widgets.dart';
```
Comment (C4s4r) [inline lib/utils/pages/feed_utils.dart:5]: Again here: In order to keep the order _dart, flutter, external packages, internal imports_ this should be moved down.
```suggestion
import 'package:flutter/widgets.dart';
import 'package:cached_network_image/cached_network_image.dart';

import 'package:campus_app/pages/rubnews/news_entity.dart';
import 'package:campus_app/pages/rubnews/widgets/feed_item.dart';
import 'package:campus_app/utils/pages/presentation_functions.dart';
```

**Review by C4s4r — CHANGES_REQUESTED** [review]
Some minor code-style changes should be made before merging this PR.

Also we could think about adding **a short article description** below the image & title. If the data source doesn't support one, we could show a shrinked version of the actual content.

File: lib/pages/rubnews/rubnews_page.dart [hunk 3]
```diff
@@ -46,10 +55,34 @@ class RubnewsPageState extends State<RubnewsPage> {
...
           if (_headerOpacity != 1) setState(() => _headerOpacity = 1);
+        } else if (_scrollController.offset < -200) {
+          // refresh data when user swipe over beginning of list
+          _rubnewsUsecases.updateFeedAndFailures().then((data) {
+            // TODO: Show loading indicator
+            setState(() {
+              _rubnews = data['news']! as List<NewsEntity>;
+              _failures = data['failures']! as List<Failure>;
+            });
+          });
```
[the remark this answers is not in the document]
Comment (domai-tb) [reply lib/pages/rubnews/rubnews_page.dart]: The performance drop was indicated by multiple calling of the `updateFeedAndFailures()`-function. I completly removed this by using the `RefreshIndicator`-widget.

**Review by C4s4r — APPROVED** [review]
Seems good to go now! 👍🏼

**Review by C4s4r — APPROVED** [review]
Looks good, ready to merge!

**Pull request merged**